src/limbo/experimental/model/spgp.hpp

```diff
@@ -85,12 +85,12 @@ namespace limbo {
                 double sig;
                 Eigen::MatrixXd xb;
 
-                HyperParams(const Eigen::VectorXd& input, const size_t& m, const size_t& dim_in) {
-                    Eigen::VectorXd w = input; // TODO: Improve
+                HyperParams(const Eigen::VectorXd& w, const size_t& m, const size_t& dim_in) {
                     b = w.segment(m*dim_in, dim_in).transpose().array().exp();
                     c = std::exp(w[(m+1)*dim_in]);
                     sig = std::exp(w[(m+1)*dim_in+1]);
-                    xb = Eigen::Map<Eigen::MatrixXd>(w.segment(0, m).data(), m, dim_in);
+                    xb = w.segment(0, m*dim_in);
+                    xb.resize(m, dim_in);
```

Comment:
Does this work? I'm thinking maybe `conservativeResize()` is more suitable for this.